src/bin/playground.cpp

```diff
@@ -408,7 +417,55 @@ file_header read_file_header(std::string filename) {
   return file_header;
 }
 
+void prepare_filestream(const char* fname) {
+  std::remove(fname);
+  FILESTREAM.open(fname, std::ios::out | std::ios::binary | std::ios::app);
+}
+
+void end_filestream() {
+  FILESTREAM.close();
+}
+
+void persist_chunks_to_disk(std::vector<std::shared_ptr<Chunk>> chunks) {
+  file_lock.acquire();
+
+  auto chunk_segment_offset_ends = std::vector<std::vector<uint32_t>>(CHUNK_COUNT);
+  auto chunk_offset_ends = std::array<uint32_t, CHUNK_COUNT>();
+  auto chunk_ids = std::array<uint32_t, CHUNK_COUNT>();
+
+  auto offset = uint32_t{sizeof(file_header)};
+  for (auto chunk_index = uint32_t{0}; chunk_index < chunks.size(); ++chunk_index) {
+    const auto segment_offset_ends = generate_segment_offset_ends(chunks[chunk_index]);
+    offset += segment_offset_ends.back();
+
+    chunk_segment_offset_ends[chunk_index] = segment_offset_ends;
+    chunk_offset_ends[chunk_index] = offset;
+  }
+  // Fill all offset fields, that are not used with 0s.
+  for (auto rest_chunk_index = chunks.size(); rest_chunk_index < CHUNK_COUNT; ++rest_chunk_index) {
```

Comment:
for runtime improvement perhaps just 0-initialize those vectors

Reply:
part of next commit :)